Content.Client/SurveillanceCamera/SurveillanceCameraMonitorSystem.cs

```diff
@@ -0,0 +1,63 @@
+namespace Content.Client.SurveillanceCamera;
+
+public sealed class SurveillanceCameraMonitorSystem : EntitySystem
+{
+    private readonly Dictionary<EntityUid, CameraSwitchTiming> _activeTimers = new();
+    private readonly List<EntityUid> _toRemove = new();
+
+    private const float InitialTime = 30;
+
+    public override void Update(float frameTime)
+    {
+        foreach (var (uid, timing) in _activeTimers)
+        {
+            timing.TimeLeft -= frameTime;
+
+            if (timing.TimeLeft <= 0)
+            {
```

Comment:
We moved most stuff to using separate components when active though apparently it has issues with prediction on clients. At the very least you'll need to consider if an entity is deleted or paused for a timer.